src/main/java/co/rsk/federate/config/PowHSMBookkeepingConfig.java

```diff
@@ -1,87 +1,86 @@
-    private long informerInterval = 2_000;
-    private boolean stopBookkeepingScheduler = false;
-    private int maxChunkSizeToHsm = 10;
 
-    public PowHSMBookkeepingConfig(SignerConfig signerConfig, String networkParameter) {
-        if (signerConfig.getConfig().hasPath("bookkeeping.difficultyTarget")) {
-            this.difficultyTarget = new BigInteger(signerConfig.getConfig().getString("bookkeeping.difficultyTarget"));
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxAmountBlockHeaders")) {
-            this.maxAmountBlockHeaders = signerConfig.getConfig().getInt("bookkeeping.maxAmountBlockHeaders");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.informerInterval")) {
-            this.informerInterval = signerConfig.getConfig().getLong("bookkeeping.informerInterval");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.stopBookkeepingScheduler")) {
-            this.stopBookkeepingScheduler = signerConfig.getConfig().getBoolean("bookkeeping.stopBookkeepingScheduler");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxChunkSizeToHsm")) {
-            this.maxChunkSizeToHsm = signerConfig.getConfig().getInt("bookkeeping.maxChunkSizeToHsm");
+    public enum NetworkDifficultyCap {
+        MAINNET(new BigInteger("7000000000000000000000")),
+        TESTNET(BigInteger.valueOf(1000000000000000L)),
+        REGTEST(BigInteger.valueOf(20L));
+
+        private final BigInteger difficultyCap;
+
+        NetworkDifficultyCap(BigInteger difficultyCap) {
+            this.difficultyCap = difficultyCap;
         }
 
-        switch (networkParameter) {
-            case NetworkParameters.ID_MAINNET:
-                this.difficultyCap = DIFFICULTY_CAP_MAINNET;
-                break;
-            case NetworkParameters.ID_TESTNET:
-                this.difficultyCap = DIFFICULTY_CAP_TESTNET;
-                break;
-            case NetworkParameters.ID_REGTEST:
-                this.difficultyCap = DIFFICULTY_CAP_REGTEST;
-                break;
-            default:
-                String message = "Invalid network specified for the Bookkeeping Config: " + networkParameter;
-                logger.error(message);
-                throw new IllegalArgumentException(message);
+        public BigInteger getDifficultyCap() {
+            return difficultyCap;
         }
     }
 
+    private static final BigInteger DEFAULT_DIFFICULTY_TARGET = BigInteger.valueOf(3);
+    private static final boolean DEFAULT_STOP_BOOKKEPING_SCHEDULER = false;
+    private static final int DEFAULT_MAX_AMOUNT_BLOCK_HEADERS = 7;
+    private static final int DEFAULT_MAX_CHUNK_SIZE = 10;
+    private static final long DEFAULT_INFORMER_INTERVAL = 2_000;
+
+    private final SignerConfig signerConfig;
+    private final String networkParameter;
+
+    public PowHSMBookkeepingConfig(SignerConfig signerConfig, String networkParameter) {
+        this.signerConfig = signerConfig;
```

Comment:
Looks like we always access `singerConfi,getConfig`. So maybe we shpuld simply store the Config object?

```suggestion
        this.signerConfig = signerConfig.getConfig();
```

Reply:
Yes I originally was going to do it that way. But since SignerConfig was more "powHsm" related I left as is. But I agree, I have updated it.